File: Parking Lot/Parking Lot Java/src/main/java/org/example/mode/FileMode.java
package org.example.mode;

import org.example.command.CommandExecutorFactory;

import java.io.*;

public class FileMode extends Mode {

    private final String fileName;

    public FileMode(CommandExecutorFactory commandExecutorFactory, String fileName) {
        super(commandExecutorFactory);
        this.fileName = fileName;
    }

    @Override
    public  void process() throws Exception {

        final File file = new File(fileName);
        final BufferedReader reader;
        try {
            reader = new BufferedReader(new FileReader(file));
        } catch (FileNotFoundException e) {
            System.out.println("Invalid file!");
            return;
        }

        String input = reader.readLine();
        while (input != null) {
            processCommand(input);
            input = reader.readLine();
        }

    }
}

File: Parking Lot/Parking Lot C#/Mode/FileMode.cs
using Commands;
using Model;

namespace Mode // Replace with your desired namespace
{
    /// <summary>
    /// Mode running in which input commands are given from a file.
    /// </summary>
    public class FileMode : Mode
    {
        private string fileName;

        public FileMode(
            CommandExecutorFactory commandExecutorFactory,
            OutputPrinter outputPrinter,
            string fileName)
            : base(commandExecutorFactory, outputPrinter)
        {
            this.fileName = fileName;
        }

        /// <inheritdoc/>
        public override void Process()
        { 
            try
            {
                using (StreamReader reader = new StreamReader(fileName))
                {
                    string input = reader.ReadLine();
                    while (input != null)
                    {
                        Command command = new Command(input);
                        ProcessCommand(command);
                        input = reader.ReadLine();
                    }
                }
            }
            catch (FileNotFoundException)
            {
                outputPrinter.InvalidFile();
            }
            catch (IOException ex)
            {
                Console.WriteLine($"An IOException occurred: {ex.Message}");
            }
        }
    }
}
